### PR TITLE
Bug 2042370: [Alibaba] fix installer index panic

### PR DESCRIPTION
**Pull request by bd233**
And add NAT gateway validation to check the region whether support NAT gateway

### CODE REVIEW

File: pkg/asset/cluster/tfvars.go [hunk 1]
```diff
@@ -741,6 +741,9 @@ func (t *TerraformVariables) Generate(parents asset.Parents) error {
 		if err != nil {
 			return errors.Wrapf(err, "failed to list avaliable zones for NAT gateway")
 		}
+		if len(natGatewayZones.Zones) == 0 {
+			return errors.Wrapf(err, "enhanced NAT gateway is not supported in the region %s", installConfig.AlibabaCloud.Region)
+		}
```
Comment (patrickdillon) [inline pkg/asset/cluster/tfvars.go:746]: Why is this needed here as well as in the machine configs? Isn't it sufficient to check once?

Comment (bd233) [reply pkg/asset/cluster/tfvars.go:746]: After I added this check, I realized it would be more appropriate to add a validation, but I'm not sure if it's superfluous, so I kept it

Comment (patrickdillon) [reply pkg/asset/cluster/tfvars.go:746]: Yeah you can remove this entire check from here. The install config validation will cover it.

Comment (bd233) [reply pkg/asset/cluster/tfvars.go]: Have removed
